src/frontends/sql/decode_tag_transform_operation_test.cc

```diff
@@ -0,0 +1,123 @@
+// limitations under the License.
+//----------------------------------------------------------------------------
+
+#include "absl/strings/str_format.h"
+#include "absl/strings/string_view.h"
+#include "google/protobuf/text_format.h"
+#include "src/common/testing/gtest.h"
+#include "src/frontends/sql/decode.h"
+#include "src/frontends/sql/decoder_context.h"
+#include "src/frontends/sql/sql_ir.pb.h"
+#include "src/frontends/sql/testing/literal_operation_view.h"
+#include "src/frontends/sql/testing/merge_operation_view.h"
+#include "src/frontends/sql/testing/tag_transform_operation_view.h"
+#include "src/frontends/sql/testing/utils.h"
+#include "src/common/utils/map_iter.h"
+
+namespace raksha::frontends::sql {
+
+namespace {
+
+using ir::Attribute;
+using ir::Block;
+using ir::IRContext;
+using ir::Operation;
+using ir::Storage;
+using ir::Value;
+using ir::types::TypeBase;
+using ir::value::OperationResult;
+using ir::value::StoredValue;
+using ::testing::Combine;
+using ::testing::Eq;
+using ::testing::IsEmpty;
+using ::testing::IsNull;
+using testing::LiteralOperationView;
+using testing::MergeOperationView;
+using ::testing::NotNull;
+using ::testing::Pair;
+using ::testing::ResultOf;
+using testing::TagTransformOperationView;
+using ::testing::TestWithParam;
+using ::testing::UnorderedElementsAre;
+using ::testing::UnorderedElementsAreArray;
+using testing::UnwrapDefaultOperationResult;
+using ::testing::Values;
+using ::testing::ValuesIn;
+using utils::MapIter;
+
+class DecodeTagTransformTest
+    : public ::testing::TestWithParam<
+          std::tuple<uint64_t, absl::string_view,
+                     std::pair<absl::string_view, std::vector<uint64_t>>>> {};
+
+TEST_P(DecodeTagTransformTest, DecodeTagTransformTest) {
+  IRContext ir_context;
+  DecoderContext decoder_context(ir_context);
+
+  // Construct the test textproto and parse it.
+  constexpr absl::string_view kTagTransformTextprotoFormat = R"(
+id: %u
+tag_transform : {
+  transform_rule_name: "%s"
+  tag_precondition_ids: [ %s ]
+  transformed_node: { %s } })";
+  const auto &[id, rule_name, value_textproto_and_ids] = GetParam();
+  const auto &[value_textproto, value_ids] = value_textproto_and_ids;
+
+  Expression expr;
+  EXPECT_TRUE(google::protobuf::TextFormat::ParseFromString(
+      absl::StrFormat(kTagTransformTextprotoFormat, id, rule_name,
+                      absl::StrJoin(value_ids, ", "), value_textproto),
+      &expr));
+  Value decoded_value = DecodeExpression(expr, decoder_context);
+
+  std::vector<Value> expected_values =
+      MapIter<Value>(value_ids, [&decoder_context](uint64_t current_id) {
+        return decoder_context.GetValue(current_id);
+      });
+
+  EXPECT_EQ(decoded_value, decoder_context.GetValue(std::get<0>(GetParam())));
+  TagTransformOperationView tag_xform_view(
+      UnwrapDefaultOperationResult(decoded_value));
+  EXPECT_EQ(tag_xform_view.GetRuleName(), std::get<1>(GetParam()));
```

Comment:
Reuse `rule_name` instead of `std::get<0>(GetParam())`?

Reply:
Fixed.